lib/authenforce/enforcers.go

```diff
@@ -1,32 +1,75 @@
 package authenforce
 
 import (
+	"context"
+	"sync/atomic"
+
 	"github.com/cuvva/cuvva-public-go/lib/cher"
-	"github.com/wearemojo/mojo-public-go/lib/authparsing"
+	"github.com/wearemojo/mojo-public-go/lib/errgroup"
+	"github.com/wearemojo/mojo-public-go/lib/merr"
 )
 
-type Enforcer func(*authparsing.AuthState) error
+type (
+	// Enforcers is a set of Enforcer functions that are run together.
+	//
+	// It is important to note that only one enforcer can acknowledge a request. If
+	// multiple enforcers ack, an error will be returned as it is unsafe. It is
+	// therefore important that your enforcers are focused to find requests that are
+	// applicable and quickly return that it won't handle the auth.
```

Comment:
Could we be a little more concrete/specific? What exactly are the rules which must be followed?